api/types.go

```diff
@@ -607,13 +611,17 @@ func DefaultOptions() Options {
 
 		Runner: Runner{
 			// options set when the model is loaded
-			NumCtx:    2048,
-			NumBatch:  512,
-			NumGPU:    -1, // -1 here indicates that NumGPU should be set dynamically
-			NumThread: 0,  // let the runtime decide
-			LowVRAM:   false,
-			UseMLock:  false,
-			UseMMap:   nil,
+			NumCtx:      2048,
+			NumBatch:    512,
+			NumGPU:      -1, // -1 here indicates that NumGPU should be set dynamically
+			NumThread:   0,  // let the runtime decide
+			LowVRAM:     false,
+			UseMLock:    false,
+			UseMMap:     nil,
+			DraftMax:    16,
+			DraftMin:    5,
+			DraftPMin:   0.9,
+			DraftNumCtx: 0, // 0 = loaded from model
```

Comment:
in api/types.go these are set to omitempty - but here they're always being set by default, perhaps they could only be passed to the runner at load time if a draft model is specified?

Reply:
Setting them (with the exception of the newly added DraftNumGPU) to defaults here is indeed not necessary! FIXED: https://github.com/ollama/ollama/pull/8134/commits/da79e2a93a8d8117066213e4c288b7401ba4eebe

If you also suggest to get speculative decoding related options entirely out of api/types.go, please open another review comment - I am open to it, I just didn't find a better place yet.